tests/test_logger.py

```diff
@@ -0,0 +1,243 @@
+    assert "DEBUG" in caplog.text
+
+    msg = f"This is an info message"
+    LOG.info(msg)
+    assert msg in caplog.text
+    assert "INFO" in caplog.text
+
+    msg = f"This is a warning message"
+    LOG.warning(msg)
+    assert msg in caplog.text
+    assert "WARNING" in caplog.text
+
+    msg = f"This is an error message"
+    LOG.error(msg)
+    assert msg in caplog.text
+    assert "ERROR" in caplog.text
+
+    msg = f"This is a critical message"
+    LOG.critical(msg)
+    assert msg in caplog.text
+    assert "CRITICAL" in caplog.text
+
+    msg = f'This is a 30 message using "log"'
+    LOG.log(30, msg)
+    assert msg in caplog.text
+
+    #  Set back to default level == ERROR
+    LOG.logger.setLevel("ERROR")
+    LOG.std_out_handler.setLevel("ERROR")
+
+
+def test_global_methods_with_log_level_warning(caplog: pytest.LogCaptureFixture):
+    """Testing global logger methods for printing out different log messages, from DEBUG to
+    CRITICAL.
+
+    Parameters
+    ----------
+    caplog : pytest.LogCaptureFixture
+        Fixture for capturing logs.
+    """
+    LOG.logger.setLevel("WARNING")
+    LOG.std_out_handler.setLevel("WARNING")
+
+    msg = f"This is a debug message"
+    LOG.debug(msg)
+    assert msg not in caplog.text
+
+    msg = f"This is an info message"
+    LOG.info(msg)
+    assert msg not in caplog.text
+
+    msg = f"This is a warning message"
+    LOG.warning(msg)
+    assert msg in caplog.text
+
+    msg = f"This is an error message"
+    LOG.error(msg)
+    assert msg in caplog.text
+
+    msg = f"This is a critical message"
+    LOG.critical(msg)
+    assert msg in caplog.text
+
+    msg = f'This is a 30 message using "log"'
+    LOG.log(30, msg)
+    assert msg in caplog.text
+
+    #  Set back to default level == ERROR
+    LOG.logger.setLevel("ERROR")
+    LOG.std_out_handler.setLevel("ERROR")
+
+
+def test_global_methods_with_log_level_critical(caplog: pytest.LogCaptureFixture):
+    """Testing global logger methods for printing out different log messages, from DEBUG to
+    CRITICAL.
+
+    Parameters
+    ----------
+    caplog : pytest.LogCaptureFixture
+        Fixture for capturing logs.
+    """
+    LOG.logger.setLevel("CRITICAL")
+    LOG.std_out_handler.setLevel("CRITICAL")
+
+    msg = f"This is a debug message"
+    LOG.debug(msg)
+    assert msg not in caplog.text
+
+    msg = f"This is an info message"
+    LOG.info(msg)
+    assert msg not in caplog.text
+
+    msg = f"This is a warning message"
+    LOG.warning(msg)
+    assert msg not in caplog.text
+
+    msg = f"This is an error message"
+    LOG.error(msg)
+    assert msg not in caplog.text
+
+    msg = f"This is a critical message"
+    LOG.critical(msg)
+    assert msg in caplog.text
+
+    msg = f'This is a 30 message using "log"'
+    LOG.log(30, msg)
+    assert msg not in caplog.text
+
+    #  Set back to default level == ERROR
+    LOG.logger.setLevel("ERROR")
+    LOG.std_out_handler.setLevel("ERROR")
+
```

Comment:
The three tests above could be combined into a single test using `pytest.mark.parametrize`. See https://docs.pytest.org/en/7.3.x/how-to/parametrize.html and similar tests in `test_additive.py`.